### DISCUSSION TITLE
Cooldown Texture Above Text

### DISCUSSION THREAD

**Discussion by croana** [SFX-WoW/Masque#200] [Issues] [answered]
After updating to the latest version today, the cooldown texture is smaller than before (this can be changed manually by increasing the cooldown size on the skin) and now covers the count and name on the button. This doesn't look that great, since both of these things often hang over the edge of the cooldown area of the button.

**Comment by StormFX** [comment, marked as answer]
_Masque_ no longer changes the frame levels of any frames, due to the potential for taint. The Cooldown covering text, etc, is the default behavior. As the change log notes, skin authors will need to update their skins accordingly.

**Comment by croana** [comment]
Sorry to be a pain, but I'm using a really old skin that isn't updated anymore.

How would I change the skin so that the cooldown DOESN'T cover the count and name of the button? Would I have to use the new custom regions somehow? Or is this literally no longer possible because 

> Masque no longer changes the frame levels of any frames

Thanks for your help and your continued work on this addon.

**Comment by StormFX** [comment]
You'd have to write a custom add-on. Each button has multiple frames that are stacked in a particular order. Those frames each contain child regions whose "z" position is limited to that of the parent frame. The only way to change a region's "z" position outside the limits of its parent is to change its parent. Think of it like a stack of boxes where each box has books in it. You can't place a book from the bottom box in between books from the boxes above it without moving the book to a different box.

To make matters worse, you can't simply move the text to one of the other frames because those frames are usually hidden when not active. What _Masque_ used to do is create an additional frame, below the button, reassign the backdrop and icon to that frame and then move the cooldown frame between this new frame and the button itself. It then had to hook and keep track of the frame level of each buttons frames to keep them in order. This was causing taint, so it was removed.

While I can understand individual preference, the current behavior is the intended behavior. If a button is on cooldown, all other regions of the button _should_ be covered by the cooldown.

**Comment by croana** [comment]
I totally got what you were saying until 

> all other regions of the button should be covered by the cooldown.

![image](https://user-images.githubusercontent.com/28045827/64739978-b6282c80-d4eb-11e9-9a08-90eae4f13ac8.png)

So this is what is happening to the count on my icons right now. It's not actually totally covered by the cooldown, because the circle of the cooldown is actually within the outside edge of the background.

By your logic, shouldn't the count (circled badly in red) actually be greyed out - all of it - and not just the part that's covered by the cooldown? Sorry if I'm being dense.

**Comment by croana** [comment]
....couldn't masque just have, like, an additional frame above the cooldown? Building upwards instead of downwards like you described in your comment? Again, sorry if I'm asking something dumb. I actually don't have very much idea what I'm talking about.

**Comment by StormFX** [comment]
> By your logic, shouldn't the count (circled badly in red) actually be greyed out - all of it - and not just the part that's covered by the cooldown? Sorry if I'm being dense.

That's because that skin probably hasn't been updated yet. It's up to skin authors to update their skins to work with _Masque_'s changes. I'm not sure what skin that is, but it looks like _Serenity_, which I haven't gotten to yet.

> ....couldn't masque just have, like, an additional frame above the cooldown? Building upwards instead of downwards like you described in your comment?

_Technically_, it could. But that defeats the point of the previous behavior being removed in the first place. Not to mention that if the Count text is above the Cooldown, it's also above the _Cooldown_'s Count text, which leads to issues [like this](https://imgur.com/EHxT6rn).

**Comment by croana** [comment]
Oh, I see what you mean. I don't use cooldown counters, so I hadn't thought of that.

I'm actually using Renaitre, which hasn't been updated in aaaaaages. I've just been making uninformed changes to it so that it doesn't break over time.  I like how the bottom of the button is a bit shaded, much like the shadow behind DK runes. Serenity looks a bit more flat and doesn't look as nice when you overlap the buttons slightly.

Idk, I guess I could change the count so that it's smaller and higher up on the button, that way the cooldown would cover it. There'd be no fix for anything that has a macro name, though, especially if the name is wider than the bottom of the circle.

I might wait and see how you tackle this problem on Serenity, since I assume it'll have the same issue??

**Comment by StormFX** [comment]
I'm going to close this as it's not something I'm going to revert. Hopefully you have or will find a solution.
